### ISSUE TITLE
convert_datetime_index_to_time does not check for index property

### ISSUE THREAD

**Issue by CommonClimate** [LinkedEarth/paleoPandas#14]
when using `from_pandas()`, an error is raised because the provided axis is not a proper datetime index. 
`convert_datetime_index_to_time()` should check for this internally and effect the appropriate conversion if the check fails. 

for full context, see Dry_Tortugas.ipynb in this repo

```
---------------------------------------------------------------------------
AttributeError                            Traceback (most recent call last)
Cell In[19], line 9
      1 metadata = {'label':'Dry Tortugas [De Long et al, 2011]',
      2             'time_name':'time',
      3             'time_unit':'yr CE',
      4             'archiveType':'coral',
      5             'lat': 24.628588, 
      6             'lon':-82.873287}
      7 ser = df.iloc[:,4].set_axis(df.iloc[:,0]) # extract S.S Sr/Ca
----> 9 ts_ss_1b = pyleo.Series.from_pandas(ser,metadata)
     10 ts_ss_1b

File ~/Documents/GitHub/Pyleoclim_util/pyleoclim/core/series.py:259, in Series.from_pandas(ser, metadata)
    258 def from_pandas(ser, metadata):
--> 259     time = tsutils.convert_datetime_index_to_time(ser.index, metadata['time_unit'], metadata['time_name'])
    260     ts = Series(value=ser.values, time=time,  
    261                       time_name = metadata['time_name'] if metadata['time_name'] is not None else ser.index.name,
    262                       time_unit = metadata['time_unit'],
   (...)
    265                       label = ser.name
    266                       )
    267     return ts

File ~/Documents/GitHub/Pyleoclim_util/pyleoclim/utils/tsutils.py:108, in convert_datetime_index_to_time(datetime_index, time_unit, time_name)
    106 else:
    107     raise ValueError(f'Expected one of {"prograde", "retrograde"}, got {direction}')
--> 108 year_diff = (datetime_index.year - int(datum))
    109 seconds_diff = (datetime_index.to_numpy() - datetime_index.year.astype(str).astype('datetime64[s]').to_numpy()).astype('int')
    110 diff = year_diff + seconds_diff / SECONDS_PER_YEAR

AttributeError: 'Float64Index' object has no attribute 'year'
```

**Comment by CommonClimate** [comment]
Apparently, when creating the time axis of `ser` the index was coerced to [Float64Index](https://pandas.pydata.org/docs/reference/api/pandas.Float64Index.html#pandas.Float64Index), which is supposed to be deprecated. 

I stubbed a strategy to check for this, but I'm wondering how we might grease the wheels so the index gets coerced into the proper type. Currently looks like this in Pyleoclim:

```
@classmethod
    def from_pandas(cls, ser, metadata):
        if isinstance(ser.index, pd.DatetimeIndex):
            time = tsutils.convert_datetime_index_to_time(ser.index, metadata['time_unit'], metadata['time_name'])
        else: # TODO: convert to proper type. 
            raise ValueError('The provided index must be a proper DatetimeIndex object')
        md = {key: metadata[key] for key in metadata.keys() if key not in {'time_name',  'value_name'}}
        return cls(
                time=time,
                value=ser.values,
                time_name =metadata['time_name'] if metadata['time_name'] is not None else ser.index.name,
                value_name=metadata['value_name'] if metadata['value_name'] is not None else ser.name,
                **md,
        )

```

**Comment by MarcoGorelli** [comment]
this looks fine, I'd suggest just raising an error (as you've done) if the index is not of the correct type

you probably also want to check that `ser.index.unit` is `'s'`, and advise the caller to convert (`.as_unit('s')`) if it's not

**Comment by CommonClimate** [comment]
Got it. After today's discussion, I did two things:
- implemented the conversion from an arbitrary time vector (e.g. decimal years) to datetime index in [tsutils](https://github.com/LinkedEarth/Pyleoclim_util/blob/new-objects/pyleoclim/utils/tsutils.py), so it can be re-used within and outside `.from_pandas()`.   @MarcoGorelli I put in "unit" as an argument here, mostly to see what would happen if we changed the units, but I have not enabled this yet because I wasn't sure of the best way to pass `astype('datetime64[unit]'). Would an f-string do?
- implemented the check on `.as_unit('s')` within `from_pandas()`:
```
def from_pandas(cls, ser, metadata):
        if isinstance(ser.index, pd.DatetimeIndex):
            index = ser.index.as_unit('s') if ser.index.unit != 's' else ser.index  
            time = tsutils.convert_datetime_index_to_time(index, metadata['time_unit'], metadata['time_name'])
        else:  
            raise ValueError('The provided index must be a proper DatetimeIndex object')
```
- tested in Dry_Tortugas.ipynb, and it works like a charm!

Will close this issue now but would appreciate feedback/help on the units business. I suggest we keep the method public for debugging purposes a tidbit longer.
